### PR TITLE
[copy.pl] Added copy of several rows

### PR DESCRIPTION
**Pull request by vague666**
feature request fulfillment #725

### CODE REVIEW

File: scripts/copy.pl [hunk 1]
```diff
@@ -102,21 +102,36 @@ sub cmd_find {
...
 	unless (defined $line) {
 		Irssi::print('No Copy!', MSGLEVEL_CLIENTCRAP);
 		return();
 	}
-	for(1..$args) {
-		my $l=$line->prev;
-		if (defined $l) {
-			$line= $l;
+
+	$args = [split /[\s-]/, $args];
```
Comment (bw1) [inline scripts/copy.pl:110]:
```
my @argv = split /[\s-]/, $args;
```

File: scripts/copy.pl [hunk 2]
```diff
@@ -7,7 +7,7 @@
 use MIME::Base64;
 use File::Glob qw/:bsd_glob/;
 
-$VERSION = '0.05';
+$VERSION = '0.10';
 %IRSSI = (
 	authors	=> 'vague,bw1',
 	contact	=> 'bw1@aol.at',
```
Comment (bw1) [inline scripts/copy.pl:13]: changed	=> '2020-09-26',

File: scripts/copy.pl [hunk 3]
```diff
@@ -102,21 +102,36 @@ sub cmd_find {
...
+	$args->[1] -= 1 if defined $args->[1];
+
+	for(1..$args->[0]) {
+		last unless $line->prev;
+		$line = $line->prev;
+        }
+
+        my $str;
+	for($args->[0]..($args->[1] // $args->[0])) {
+		$str = join "\n", $line->get_text(0), $str;
```
Comment (bw1) [inline scripts/copy.pl:127]: settings copy_eol  ?

File: scripts/copy.pl (hunk 1 above)
Comment (vague666) [reply scripts/copy.pl:110]: I prefer references :P but I'll update!

File: scripts/copy.pl [hunk 4]
```diff
@@ -107,31 +107,30 @@ sub cmd_num {
 		return();
 	}
 
-	$args = [split /[\s-]/, $args];
-	if(@$args > 2 || grep {/[^\d]+/} @$args) {
+	my @arg = [split /[\s-]/, $args];
```
Comment (bw1) [inline scripts/copy.pl:110]: sorry, no ref

File: scripts/copy.pl [hunk 5]
```diff
@@ -107,31 +107,30 @@ sub cmd_num {
...
+	for(1..$arg[0]) {
 		last unless $line->prev;
 		$line = $line->prev;
         }
 
         my $str;
-	for($args->[0]..($args->[1] // $args->[0])) {
-		$str = join "\n", $line->get_text(0), $str;
+	for($arg[0]..($arg[1] // $arg[0])) {
+		$str = join +(Irssi::settings_get_str($IRSSI{name}.'_file_eol') || "\n"), $line->get_text(0), $str;
```
Comment (bw1) [inline scripts/copy.pl:127]: Irssi::settings_get_str($IRSSI{name}.'_file_eol') == '\n'

$copy_file_eol == "\n"

**Comment by bw1** [comment]
```
127 		$str = join +(Irssi::settings_get_str($IRSSI{name}.'_file_eol') || "\n"), $line->get_text(0), $str;
```

```
273 	my $fe= Irssi::settings_get_str($IRSSI{name}.'_file_eol');
274 	$fe =~ s/\\n/\n/g;
275 	$fe =~ s/\\t/\t/g;
276 	$copy_file_eol= $fe;
```

```
285 Irssi::settings_add_str($IRSSI{name} ,$IRSSI{name}.'_file_eol', '\n');
```

**Comment by bw1** [comment]
<https://github.com/irssi/scripts.irssi.org/pull/726#discussion_r495576005>

<https://github.com/irssi/scripts.irssi.org/pull/726#issuecomment-699661316>

File: scripts/copy.pl (hunk 5 above)
Comment (vague666) [reply scripts/copy.pl:127]: I missed the global, fixed

**Pull request merged**